homeassistant/components/melcloud/climate.py

```diff
@@ -116,6 +141,30 @@ def name(self):
         """Return the display name of this entity."""
         return self._name
 
+    @property
+    def device_state_attributes(self) -> Optional[Dict[str, Any]]:
+        """Return the optional state attributes with device specific additions."""
+        attr = {}
+
+        vane_horizontal = self._device.vane_horizontal
+        if vane_horizontal:
+            attr.update(
+                {
+                    ATTR_VANE_HORIZONTAL: vane_horizontal,
+                    ATTR_VANE_HORIZONTAL_POSITIONS: self._device.vane_horizontal_positions,
```

Comment:
Please create an attribute for self._device.vane_horizontal_positions so it can be used in the service validation.